conans/client/generators/cmake_common.py

```diff
@@ -826,7 +826,7 @@ class CMakeCommonMacros:
     """)
 
     apple_frameworks_macro = textwrap.dedent("""
-        macro(conan_find_apple_frameworks FRAMEWORKS_FOUND FRAMEWORKS SUFFIX)
+        macro(conan_find_apple_frameworks FRAMEWORKS_FOUND FRAMEWORKS SUFFIX BUILD_TYPE)
             if(APPLE)
```

Comment:
Can we move the logic here to use a helper variable, and use the same block below?

```cmake
if(CMAKE_BUILD_TYPE)
    set(_BTYPE CMAKE_BUILD_TYPE)
elseif(NOT BUILD_TYPE STREQUAL "")
    set(_BTYPE BUILD_TYPE)
endif()

```

Reply:
I don't know, the code is way too complicated and too fragile, I'd avoid touching it too much to reduce breakage. 
as I can say, variables have different meaning/values, and it might be `_DEBUG` or `Debug`, so they couldn't be used interchangeably that easy.